content/chat/rooms/messages.textile

```diff
@@ -126,3 +146,235 @@ const MyComponent = () => {
+
+```[javascript]
+import { Message } from '@ably/chat';
+const messageToUpdate: Message
+await room.messages.update(messageToUpdate, { "text": "my updated text" }, { description: 'Message update by user' });
+```
+
+```[react]
+import { useMessages } from '@ably/chat/react';
+import { Message } from '@ably/chat';
+
+const MyComponent = () => {
+  const { update } = useMessages();
+  const [message, setMessage] = useState<Message>();
+
+  const handleMessageUpdate = (msg: Message) => {
+  const newText = prompt('Enter new text');
+      if (!newText) {
+        return;
+      }
+    update(msg, { "text", newText }, { description: 'Message update by user' });
+  };
+
+  return (
+    <div>
+      <button onClick={() => handleMessageUpdate(message)}>Update Message</button>
+    </div>
+  );
+};
+```
+
+h3(#filter-updates). Filter for updates
+
+blang[javascript].
+  Use the "@messages.subscribe()@":https://sdk.ably.com/builds/ably/ably-chat-js/main/typedoc/interfaces/chat_js.Messages.html#subscribe method to receive messages in a room. To filter for updated messages, provide a listener that checks the @type@ property of the message event:
+
+blang[react].
+  Use the "@useMessages@":https://sdk.ably.com/builds/ably/ably-chat-js/main/typedoc/functions/chat_react.useMessages.html hook to subscribe to messages in a room. To filter for updated messages, provide a listener that checks the @type@ property of the message event:
+
+```[javascript]
+import { MessageEvents } from '@ably/chat';
+const {unsubscribe} = room.messages.subscribe((event) => {
+  switch (event.type) {
+    case MessageEvents.Created:
+      console.log('Received message: ', event.message);
+      break;
+    case MessageEvents.Updated:
+      console.log('Message updated: ', event.message);
+      break;
+    default:
+      break;
+  }
+});
+```
+
+```[react]
+import { useMessages } from '@ably/chat/react';
+import { MessageEvents } from '@ably/chat';
+
+const MyComponent = () => {
+  useMessages({
+    listener: (event) => {
+      switch (event.type) {
+        case MessageEvents.Created:
+          console.log('Received message: ', event.message);
+          break;
+        case MessageEvents.Updated:
+          console.log('Message updated: ', event.message);
+          break;
+        default:
+          break;
+      }
+    },
+  });
+
+  return <div>...</div>;
+};
+```
+
+h3(#deletion-structure). Message update structure
+
+The following is the structure of an updated message:
+
+```[json]
+{
+  "serial": "01726232498871-001@abcdefghij:001",
+  "clientId": "basketLover014",
+  "roomId": "basketball-stream",
+  "text": "What a shot! Edit: I meant to say 'What a dunk!'",
+  "headers": {},
+  "metadata": {},
+  "createdAt": "2024-06-12T11:37:59.988Z",
+  "action": "message.update",
+  "version": "01826232498871-001@abcdefghij:001",
+  "timestamp":"2024-11-21T15:49:25.425Z",
```

Comment:
Always a unix timestamp over the wire, not sure what's the _correct_ way to describe a Date instance here.

Should we change the language to `[javascript]` and use this:
```suggestion
  "timestamp": new Date("2024-11-21T15:49:25.425Z"),
```

Same for `createdAt`.

Reply:
Sure, I'll make the change :)